Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -48,6 +48,7 @@
 
 @implementation NSUserDefaults {
     StrongId<NSMutableDictionary> _registrationDict;
+    StrongId<NSOperationQueue> _synchronizeQueue;
 }
```

Comment:
I'm somewhat wary of introducing a dependency on NSOperationQueue here; there might be some way to do this with dispatch groups or another primitive.
NSOperationQueue could spawn its own thread pool to service requests like this, but we don't need an entire thread sitting around dedicated to saving user preferences.